types/filter.d.ts

```diff
@@ -10,11 +10,16 @@ export function filter<A, P extends A>(
   <B extends A>(list: readonly B[]): P[];
 };
 
-// filter(() => boolean)
+// filter(() => boolean, list)
 export function filter<T>(
   pred: (value: T) => boolean,
 ): <P extends T, C extends readonly P[] | Record<string, P>>(collection: C) => C;
 
+// filter(() => narrow, map)
+export function filter<K, T, P extends T>(pred: (val: T) => val is P, map: readonly Map<K, T>): Map<K, P>;
+// filter(() => boolean, map)
+export function filter<K, T>(pred: (val: T) => boolean, map: readonly Map<K, T>): Map<K, T>;
```

Comment:
'readonly' type modifier is only permitted on array and tuple literal types.

```suggestion
export function filter<K, T, P extends T>(pred: (val: T) => val is P, map: Map<K, T>): Map<K, P>;
// filter(() => boolean, map)
export function filter<K, T>(pred: (val: T) => boolean, map: Map<K, T>): Map<K, T>;
```